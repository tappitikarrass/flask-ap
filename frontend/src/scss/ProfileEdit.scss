@use "base16" as *;

// desktop
@media screen and (min-width: 810px) {
  #profile-edit.content {
    width: 40vw;
    margin-left: 30vw;
  }
}

// tablet
@media screen and (min-width: 500px) and (max-width: 810px) {
  #profile-edit.content {
    width: 60vw;
    margin-left: 20vw;
  }
}
