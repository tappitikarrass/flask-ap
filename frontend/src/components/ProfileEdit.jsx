import React, { useState, useRef, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useCookies } from 'react-cookie';
import { useForm } from 'react-hook-form';
import '../scss/ProfileEdit.scss';

function ProfileEdit() {
  const navigate = useNavigate();
  const [cookies] = useCookies(['token', 'username']);
  const editForm = useRef();
  const [userData, setUserData] = useState([]);

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  useEffect(() => {
    async function effect() {
      const response = await fetch(
        `http://localhost/backend/user/${cookies.username}`,
        {
          method: 'GET',
          headers: {
            Authorization: `Bearer ${cookies.token}`,
          },
        },
      );
      setUserData(await response.json());
    }
    effect();
  });

  async function onSubmit() {
    const updateData = new FormData(editForm.current);
    const updateJson = JSON.stringify(Object.fromEntries(updateData));

    const response = await fetch(
      `http://localhost/backend/user/${cookies.username}`,
      {
        method: 'PUT',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${cookies.token}`,
        },
        body: updateJson,
      },
    );
    if (response.status === 200) {
      navigate('/profile');
    }
  }

  return (
    <div id="profile-edit" className="content">
      <h2>Edit Profile</h2>
      <form
        id="edit-form"
        ref={editForm}
        method="post"
        acceptCharset="utf-8"
        onSubmit={handleSubmit(onSubmit)}
      >
        <div className="ui-row">
          <input
            type="text"
            id="firstname"
            {...register('firstname', { required: 'First name is required', maxLength: 256 })}
            placeholder="First name"
            className="field"
            defaultValue={userData.firstname}
          />
          <input
            type="text"
            id="lastname"
            {...register('lastname', { required: 'Last name is required', maxLength: 256 })}
            placeholder="Last name"
            className="field-end"
            defaultValue={userData.lastname}
          />
        </div>
        <p>{errors.firstname?.message}</p>
        <p>{errors.lastname?.message}</p>
        <input
          type="text"
          id="email"
          {...register('email', { required: 'Email is required', maxLength: 256 })}
          placeholder="Email"
          className="field"
          defaultValue={userData.email}
        />
        <p>{errors.phone?.email}</p>
        <input
          type="text"
          id="phone"
          {...register('phone', {
            required: 'Phone is required',
            maxLength:
            {
              value: 16,
              message: 'Max phone length is 16 numbers.',
            },
          })}
          placeholder="Phone"
          className="field"
          defaultValue={userData.phone}
        />
        <p>{errors.phone?.message}</p>
        <input
          type="password"
          id="password"
          {...register('password', {
            required: 'Password is required',
            minLength:
            {
              value: 8,
              message: 'Min password length is 8 characters.',
            },
            maxLength:
            {
              value: 64,
              message: 'Max password length is 64 characters.',
            },
          })}
          placeholder="New password"
          className="field"
        />
        <p>{errors.password?.message}</p>
        <div className="ui-row">
          <button type="submit" className="bt-blue">
            Save
          </button>
          <button
            type="submit"
            className="bt-red-end"
            onClick={() => navigate('/profile')}
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  );
}

export default ProfileEdit;
