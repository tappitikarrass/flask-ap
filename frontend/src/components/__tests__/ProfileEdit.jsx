import React from "react";
import { HashRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import ProfileEdit from '../ProfileEdit';

describe('ProfileEdit', () => {
  it('',() => {
    let component;
    act(() => {
      component = create(
        <HashRouter>
          <ProfileEdit />
        </HashRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
