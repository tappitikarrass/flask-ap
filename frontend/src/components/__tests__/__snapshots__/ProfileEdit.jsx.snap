// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileEdit  1`] = `
<div
  className="content"
  id="profile-edit"
>
  <h2>
    Edit Profile
  </h2>
  <form
    acceptCharset="utf-8"
    id="edit-form"
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="ui-row"
    >
      <input
        className="field"
        id="firstname"
        name="firstname"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="First name"
        type="text"
      />
      <input
        className="field-end"
        id="lastname"
        name="lastname"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Last name"
        type="text"
      />
    </div>
    <p />
    <p />
    <input
      className="field"
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Email"
      type="text"
    />
    <p />
    <input
      className="field"
      id="phone"
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Phone"
      type="text"
    />
    <p />
    <input
      className="field"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="New password"
      type="password"
    />
    <p />
    <div
      className="ui-row"
    >
      <button
        className="bt-blue"
        type="submit"
      >
        Save
      </button>
      <button
        className="bt-red-end"
        onClick={[Function]}
        type="submit"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
